pkg/services/ngalert/notifier/file_store.go

```diff
@@ -62,6 +63,46 @@ func (fileStore *FileStore) FilepathFor(ctx context.Context, filename string) (s
 	return fileStore.pathFor(filename), err
 }
 
+// GetFullState returns a base64-encoded string containing the Alertmanager's silences and nflog.
+func (fileStore *FileStore) GetFullState(ctx context.Context) (string, error) {
```

Comment:
I think we need a better description but I'll wait for a suggestion once I see the final implementation.